### PR TITLE
update pasteup and bring order to pillars

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?
Pasteup now uses the new palette and most of the places we use pillars now are a bit more strongly typed. We should create `PillarType` instead of `LinkType` when reading a pillar in now, but this should wait until @MatthewJWalls 's config pr.

## Why?
To use the new colours and make thinks a little easier to reason about.

### CODE REVIEW

File: frontend/components/Header/Nav/Pillars/pillars.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    'lifestyle',
+];
+
+export const pillarPalette: { [K in Pillar]: PillarColours } = {
+    news: palette.news,
+    opinion: palette.opinion,
+    sport: palette.sport,
+    culture: palette.culture,
+    lifestyle: palette.lifestyle,
+};
```
Comment (SiAdcock) [inline frontend/components/Header/Nav/Pillars/pillars.ts]: Why not keep these in `pasteup`?

File: frontend/components/Header/Nav/Pillars/pillars.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export const pillarMap: <T>(
+    f: (name: Pillar) => T,
+) => { [K in Pillar]: T } = f => ({
+    news: f('news'),
+    opinion: f('opinion'),
+    sport: f('sport'),
+    culture: f('culture'),
+    lifestyle: f('lifestyle'),
+}); // We could do this with an Object.assign, but that would need weirdness in ts and is less readable.
```
Comment (SiAdcock) [inline frontend/components/Header/Nav/Pillars/pillars.ts]: This is really difficult to follow. I would much rather disable the type checking that is causing a problem here and set [`suppressImplicitAnyIndexErrors`](https://github.com/Microsoft/TypeScript/issues/1232#issuecomment-64510362) to `true` in the [`tsconfig`](https://www.typescriptlang.org/docs/handbook/compiler-options.html).

File: frontend/components/Header/Nav/Pillars/pillars.ts (hunk 1 above)
Comment (AWare) [reply frontend/components/Header/Nav/Pillars/pillars.ts]: I thought about this, but I think the modelling of pillars doesn't belong in pasteup.  

The pillars in pasteup are defined by the <a href="https://design.theguardian.com/#colour-palette-4">design guide.</a>

Whereas the pillars in pillars.ts are defined by our split of content into pillars. My aim for this is to only ever check once whether a given value of pillar is actually a pillar by having every pillar reference use these types. I probably should move this file to somewhere more sensible than the nav though.

Comment (SiAdcock) [reply frontend/components/Header/Nav/Pillars/pillars.ts]: Sounds good, thanks for the explanation. Shall we move this up to the top level `frontend` directory for now?

Comment (AWare) [reply frontend/components/Header/Nav/Pillars/pillars.ts]: Moved. 👍

**Pull request merged**